control-plane/subcommand/server-acl-init/command_test.go

```diff
@@ -2238,6 +2221,431 @@ func TestRun_GatewayErrors(t *testing.T) {
+				AuthMethod:  authMethodName,
+				BearerToken: jwtToken,
+				Meta:        map[string]string{},
+			}, &api.WriteOptions{})
+			require.NoError(t, err)
+
+			require.Equal(t, len(tok.Roles), len(c.Roles))
+			for _, role := range tok.Roles {
+				require.Contains(t, c.Roles, role.Name)
+			}
+			require.Equal(t, !c.GlobalToken, tok.Local)
+		})
+	}
+}
+
+// Test that the component auth method gets created.
+func TestRun_PrimaryDatacenter_ComponentAuthMethod(t *testing.T) {
+	t.Parallel()
+
+	k8s, testSvr := completeSetup(t)
+	setUpK8sServiceAccount(t, k8s, ns)
+	defer testSvr.Stop()
+
+	// Run the command.
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+	cmdArgs := []string{
+		"-timeout=1m",
+		"-k8s-namespace=" + ns,
+		"-server-address", strings.Split(testSvr.HTTPAddr, ":")[0],
+		"-server-port", strings.Split(testSvr.HTTPAddr, ":")[1],
+		"-resource-prefix=" + resourcePrefix}
+
+	responseCode := cmd.Run(cmdArgs)
+	require.Equal(t, 0, responseCode, ui.ErrorWriter.String())
+
+	// Check that the expected policy was created.
+	bootToken := getBootToken(t, k8s, resourcePrefix, ns)
+	consulClient, err := api.NewClient(&api.Config{
+		Address: testSvr.HTTPAddr,
+		Token:   bootToken,
+	})
+	require.NoError(t, err)
+	authMethod, _, err := consulClient.ACL().AuthMethodRead(resourcePrefix+"-k8s-component-auth-method", &api.QueryOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, authMethod)
+}
+
+// Test that the local and global component auth methods gets created when run in the
+// secondary datacenter.
+func TestRun_SecondaryDatacenter_ComponentAuthMethod(t *testing.T) {
+	t.Parallel()
+
+	bootToken := "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee"
+	tokenFile := common.WriteTempFile(t, bootToken)
+	k8s, consul, consulHTTPAddr, cleanup := mockReplicatedSetup(t, bootToken)
+	setUpK8sServiceAccount(t, k8s, ns)
+	defer cleanup()
+
+	// Run the command.
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+	cmdArgs := []string{
+		"-federation",
+		"-timeout=1m",
+		"-k8s-namespace=" + ns,
+		"-auth-method-host=" + "https://my-kube.com",
+		"-acl-replication-token-file", tokenFile,
+		"-server-address", strings.Split(consulHTTPAddr, ":")[0],
+		"-server-port", strings.Split(consulHTTPAddr, ":")[1],
+		"-resource-prefix=" + resourcePrefix}
+
+	responseCode := cmd.Run(cmdArgs)
+	require.Equal(t, 0, responseCode, ui.ErrorWriter.String())
+
+	// Check that the expected local auth-method was created.
+	localAuthMethod, _, err := consul.ACL().AuthMethodRead(resourcePrefix+"-k8s-component-auth-method", &api.QueryOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, localAuthMethod)
+	// Check that the expected global auth-method was created.
+	globalAuthMethod, _, err := consul.ACL().AuthMethodRead(resourcePrefix+"-k8s-component-auth-method-dc2", &api.QueryOptions{Datacenter: "dc1"})
+	require.NoError(t, err)
+	require.NotNil(t, globalAuthMethod)
```

Comment:
does it make sense to assert that the globalAuthMethod's datacenter is correct, or are we cool bc of the QueryOption?

Reply:
The query option should be sufficient here! good question though.